File: problems/HackerRank/2521_exercises/tree/lowestCommonAncestor.c
struct TreeNode *lowestCommonAncestor(struct TreeNode *root, struct TreeNode *p, struct TreeNode *q)
{

	struct TreeNode *cur = root;

	while (cur)
	{
		if (p->val > cur->val && q->val > cur->val)
		{
			cur = cur->right;
		}
		else if (p->val < cur->val && q->val < cur->val)
		{
			cur = cur->left;
		}
		else
		{
			return cur;
		}
	}
	return cur;
}

File: problems/LeetCode/tree/lowestCommonAncestor.py
def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
		if not root:
				return None
		
		if root is p or root is q:
				return root
		
		l = self.lowestCommonAncestor(root.left, p, q)
		r = self.lowestCommonAncestor(root.right, p, q)

		if l and r:
				return root
		
		return l or r